### PR TITLE
Adds code coverage workflow to CI

### PR DESCRIPTION
**Pull request by saibatizoku**
Adds workflow to get code coverage information using `cargo-nextest` and `cargo-llvm-cov` in a github action that publishes to coveralls.io.

### CODE REVIEW

**Comment by minikin** [comment]
@saibatizoku looks like something is wrong with this coverage.

**Comment by saibatizoku** [comment]
> @saibatizoku looks like something is wrong with the coverage flow.

Yeah, @minikin I'm working on fixing this.

File: .github/workflows/ci-tests-n-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+name: Testing and Code Coverage
+
+on:
```
Comment (minikin) [inline .github/workflows/ci-tests-n-coverage.yml:3]: This a friendly reminder that yesterday we agreed to tun this job nightly on main.

**Review by minikin — APPROVED** [review]
Approved with a small changes request.

File: .github/workflows/ci-tests-n-coverage.yml (hunk 1 above)
Comment (saibatizoku) [reply .github/workflows/ci-tests-n-coverage.yml:3]: Sorry, had forgotten to add this. Thanks.

**Review by minikin — APPROVED** [review]

**Pull request merged**